src/lib/storage/table.cpp

```diff
@@ -150,7 +150,7 @@ void Table::append_chunk(const Segments& segments, const std::optional<Polymorph
 #if HYRISE_DEBUG
   for (const auto& segment : segments) {
     DebugAssert(segment->size() == chunk_size, "Segments don't have the same length");
-    const auto is_reference_segment = std::dynamic_pointer_cast<ReferenceSegment>(segment) != nullptr;
```

Comment:
For something like this, I actually don't like using the automatic conversion. The name `is_reference_segment` clearly implies a boolean, but the actual type of the variable is a pointer. So I would either rename it to just `reference_segment`, or keep the `!= nullptr`. I prefer the latter, because we actually only want a boolean, and don't use the pointer even if it is valid.

Reply:
the bane of Search&Replace...